src/main/java/com/rackspacecloud/metrics/influxdbscaler/collectors/InfluxDBHelper.java

```diff
@@ -57,27 +57,30 @@ public String getMinInstance(Collection<String> influxDBInstances) throws Except
         return minInstance;
     }
 
-    public Map<String, StatsResults.SeriesMetric[]> getSeriesMetricCollection(
+    public ConcurrentMap<String, StatsResults.SeriesMetric[]> getSeriesMetricCollection(
             final Collection<String> instances) throws Exception {
 
-        Map<String, StatsResults.SeriesMetric[]> seriesMetricCollectionMap = new HashMap<>();
+        ConcurrentMap<String, StatsResults.SeriesMetric[]> seriesMetricCollectionMap = new ConcurrentHashMap<>();
+
+        ConcurrentMap<String, Future<StatsResults>> futureResults = new ConcurrentHashMap<>();
 
+        // Query InfluxDB instance stats for ALL of the instances concurrently
         for(String baseUrl : instances) {
-            // Get all of the stats for given InfluxDB instance
-            StatsResults result = getResponseEntity(baseUrl, SHOW_STATS);
+            futureResults.put(baseUrl, executorService.submit(() -> getResponseEntity(baseUrl, SHOW_STATS)));
```

Comment:
getResponseEntity should use bounded retries on failure (not sure if it does so now). This is because large amounts of simultaneous connections are a bit more stressing, and we might want to push the thread count past 10 at some point.

Reply:
Thread count is configurable. I am using `10` for now.